rclcpp/src/rclcpp/executors/static_single_threaded_executor.cpp

```diff
@@ -50,6 +51,34 @@ StaticSingleThreadedExecutor::spin()
   }
 }
 
+void
+StaticSingleThreadedExecutor::spin_some(std::chrono::nanoseconds max_duration)
+{
+  // Make sure the entities collector has been initialized
+  if (!entities_collector_->is_init()) {
+    throw std::runtime_error("spin_some() called without initializing the entities collector");
```

Comment:
couldn't this implicitly initialize the entities collector?
Other executors don't need an initialization step before calling `spin_some()`.

Reply:
Yes, it can. The new implementation implicitly initializes the entities collector